src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -929,17 +955,17 @@ public SearchPublisher getSearch() {
 
   @Override
   public void setLocale(String code) {
-    this.locale = new Locale(code);
+    Runtime.getInstance().setLocale(code);
   }
 
   @Override
   public String getLanguage() {
-    return locale.getLanguage();
+    return Runtime.getInstance().getLanguage();

```

Comment:
This should be the language from the bot that is currently loaded.  I don't think this is right.

Reply:
This language should be derived from the locale property setting for the current bot.   not from runtime.

Reply:
this has been pushed up to Service .. and all services use their local property - if not set then its set from runtime